.github/workflows/tpu-testing.yml

```diff
@@ -0,0 +1,105 @@
+
+    # Setup gcloud CLI.
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+      with:
+        version: '290.0.1'
+        service_account_key: ${{ secrets.GKE_SA_KEY_BASE64 }}
+        project_id: ${{ secrets.GKE_PROJECT }}
+        export_default_credentials: true
+
+    # Configure Docker to use the gcloud command-line tool as a credential
+    # helper for authentication.
+    - run: |-
+        gcloud --quiet auth configure-docker
+    - name: Build and Push Docker Image
+      run: |
+        cd main/docker/tpu
+        docker build --tag "$IMAGE:$GITHUB_RUN_ID" -f Dockerfile --build-arg "GITHUB_REF=$GITHUB_REF" --build-arg "TEST_IMAGE=1" .
+        docker push "$IMAGE:$GITHUB_RUN_ID"
+
+    # Install jsonnet.
+    - run: |-
+        go get github.com/google/go-jsonnet/cmd/jsonnet
+    # Get the GKE credentials so we can deploy to the cluster
+    # Use either zone or region depending on cluster setup.
+    - run: |-
+        gcloud container clusters get-credentials "$GKE_CLUSTER" --zone "$GKE_ZONE"
+
+    - run: |-
+        # Deploy the job on the kubernetes cluster.
+        job_name=$(jsonnet -J ml-testing-accelerators/ main/docker/tpu/tpu_test-cases.jsonnet --ext-str image=$IMAGE --ext-str image-tag=$GITHUB_RUN_ID | kubectl create -f -) && \
+        job_name=${job_name#job.batch/} && \
+        job_name=${job_name% created} && \
+        echo "Waiting on kubernetes job: $job_name in cluster: $GKE_CLUSTER" && \
+        i=0 && \
+        max_checks=10 && \
+        status_code=2 && \
+        # Check on the job periodically. Set the status code depending on what
+        # happened to the job in Kubernetes. If we try max_checks times and
+        # still the job hasn't finished, give up and return the starting
+        # non-zero status code.
+        while [ $i -lt $max_checks ]; do ((i++)); if kubectl get jobs $job_name -o jsonpath='Failed:{.status.failed}' | grep "Failed:1"; then status_code=1 && break; elif kubectl get jobs $job_name -o jsonpath='Succeeded:{.status.succeeded}' | grep "Succeeded:1" ; then status_code=0 && break; else echo "Job not finished yet"; fi; sleep 30; done && \
+        echo "Done waiting. Job status code: $status_code" && \
+        # Allow time for logs to flush.
+        sleep 60 && \
+        echo "JOB_NAME: $job_name" && \
+        echo "GKE_CLUSTER: $GKE_CLUSTER" && \
+        echo "GKE_ZONE: $GKE_ZONE" && \
+        gcloud logging read "resource.type=k8s_container resource.labels.project_id=$PROJECT_ID resource.labels.location=$GKE_ZONE resource.labels.cluster_name=$GKE_CLUSTER resource.labels.namespace_name=default resource.labels.pod_name:$job_name" --limit 10000000 --order asc --format 'value(textPayload)' --project=$PROJECT_ID > /tmp/full_output.txt && \
+        csplit /tmp/full_output.txt '/<?xml version="1.0" ?>/' && \
+        # First portion is the test logs. Print these to Github Action stdout.
+        cat xx00 && \
```

Comment:
later you upload `./xx01` so what is the relation?

Reply:
xx00 is the first half of the `csplit` which contains test logs and xx01 is the second half which contains the coverage XML. These are the default filenames of the `csplit` command